lib/ios/app.nit

```diff
@@ -0,0 +1,187 @@
+
+	- (BOOL)application:(UIApplication *)application
+		willFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+
+		// Set aside `application` to be used from Nit
+		App_ui_application__assign(app_nit_ios_app, application);
+		App_app_delegate__assign(app_nit_ios_app, self);
+
+		// Complete the callback
+		return App_will_finish_launching_with_options(app_nit_ios_app);
+	}
+
+	- (BOOL)application:(UIApplication *)application
+		didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+
+		return App_did_finish_launching_with_options(app_nit_ios_app);
+	}
+
+	- (void)applicationWillResignActive:(UIApplication *)application {
+		App_will_resign_active(app_nit_ios_app);
+	}
+
+	- (void)applicationDidEnterBackground:(UIApplication *)application {
+		App_did_enter_background(app_nit_ios_app);
+	}
+
+	- (void)applicationWillEnterForeground:(UIApplication *)application {
+		App_will_enter_foreground(app_nit_ios_app);
+	}
+
+	- (void)applicationDidBecomeActive:(UIApplication *)application {
+		App_did_become_active(app_nit_ios_app);
+	}
+
+	- (void)applicationWillTerminate:(UIApplication *)application {
+		App_will_terminate(app_nit_ios_app);
+	}
+
+	@end
+`}
+
+# Application interface to the iOS system
+extern class AppDelegate in "ObjC" `{ AppDelegate * `}
+end
+
+# Graphical application to which events are sent
+extern class UIApplication in "ObjC" `{ UIApplication * `}
+end
+
+redef class App
+
+	# Main graphical application
+	var ui_application: UIApplication
+
+	# Application interface to the iOS system
+	var app_delegate: AppDelegate
+
+	# Copy back to C the command line arguments
+	#
+	# Nit extracts the first arguments from the `args` sequence,
+	# so we need to add it back. That's why Nit's `args` is smaller than in C.
+	private fun register_args(program_name: NativeString, argc: Int,
+	argv: Sequence[String]) import Sequence[String].[], String.to_cstring in "ObjC" `{
+		app_nit_ios_argc = argc+1;
+
+		// TODO copy or pin the strings when needed
+		app_nit_ios_argv = malloc(argc * sizeof(char*));
+		app_nit_ios_argv[0] = program_name;
+		for (int i = 0; i < argc; i ++) {
+			String arg = Sequence_of_String__index(argv, i);
+			app_nit_ios_argv[i+1] = String_to_cstring(arg);
+		}
+	`}
+
+	# Register `self` globally in C so it can be retrieved from iOS callbacks
+	private fun register_globally in "ObjC" `{
+		App_incr_ref(recv);
+		app_nit_ios_app = recv;
+	`}
+
+	# Entry point to the iOS framework
+	private fun ui_application_main: Bool import did_finish_launching_with_options,
+	will_finish_launching_with_options,
+	will_resign_active, did_enter_background, will_enter_foreground,
+	did_become_active, will_terminate, ui_application=, app_delegate= in "ObjC" `{
+
+		@autoreleasepool {
+			return UIApplicationMain(app_nit_ios_argc, app_nit_ios_argv,
+				nil, NSStringFromClass([AppDelegate class]));
+		}
+	`}
+
+	# Very first custom code to be executed
+	fun will_finish_launching_with_options: Bool do return true
```

Comment:
Yep they are callback, with the corresponding name in Objective-C. And I do plan on unifying these services with app.nit.

With the compilation I will be able to test all the callbacks and complete the documentation. I also plan on putting more effort in describing how these callbacks relate to app.nit's. I don't want a programmer to uselessly use platform specific callback when they could as well use those of app.nit.